File: sdk/samples/WCFSamples/TechnologySamples/Scenario/WorkflowServices/Conversations/VB/Customer/Contracts.vb
' Copyright (c) Microsoft Corporation. All rights reserved.

Imports System
Imports System.Collections.Generic
Imports System.Runtime.Serialization
Imports System.ServiceModel
Imports System.Text
Imports System.Xml

Namespace Microsoft.WorkflowServices.Samples
	<ServiceContract()> _
	Public Interface IOrder
        <OperationContract(IsInitiating:=True)> _
        Function SubmitOrder(ByVal po As PurchaseOrder, ByVal context As IDictionary(Of String, String)) As String
	End Interface

	<ServiceContract()> _
	Public Interface IOrderDetails
		<OperationContract()> _
		Function OrderDetails(ByVal po As PurchaseOrder, ByVal quote As ShippingQuote) As String
	End Interface

	<ServiceContract()> _
	Public Interface IShippingRequest
        <OperationContract(IsInitiating:=True)> _
        Function RequestShippingQuote(ByVal po As PurchaseOrder, ByVal context As IDictionary(Of String, String)) As String
	End Interface

	<ServiceContract()> _
	Public Interface IShippingQuote
		<OperationContract(IsOneWay := True)> _
		Sub ShippingQuote(ByVal quote As ShippingQuote)
	End Interface

	<DataContract()> _
	Public Class PurchaseOrder
		<DataMember()> _
		Public OrderId As Integer
		<DataMember()> _
		Public Amount As Integer
		<DataMember()> _
		Public CustomerName As String
	End Class

	<DataContract()> _
	Public Class ShippingQuote
		<DataMember()> _
		Public ShippingCost As Integer
		<DataMember()> _
		Public EstimatedShippingDate As DateTime
	End Class
End Namespace

File: sdk/samples/WCFSamples/TechnologySamples/Scenario/WorkflowServices/Conversations/CS/Customer/Contracts.cs
//  Copyright (c) Microsoft Corporation. All rights reserved.

using System;
using System.Collections.Generic;
using System.Runtime.Serialization;
using System.ServiceModel;
using System.Xml;

namespace Microsoft.WorkflowServices.Samples
{
    [ServiceContract]
    public interface IOrder
    {
        [OperationContract(IsInitiating=true)]
        string SubmitOrder(PurchaseOrder po, IDictionary<string,string> context);
    }

    [ServiceContract]
    public interface IOrderDetails
    {
        [OperationContract]
        string OrderDetails(PurchaseOrder po, ShippingQuote quote);
    }

    [ServiceContract]
    public interface IShippingRequest
    {
        [OperationContract(IsInitiating = true)]
        string RequestShippingQuote(PurchaseOrder po, IDictionary<string, string> context);
    }

    [ServiceContract]
    public interface IShippingQuote
    {
        [OperationContract(IsOneWay = true)]
        void ShippingQuote(ShippingQuote quote);
    }

    [DataContract]
    public class PurchaseOrder
    {
        [DataMember]
        public int OrderId;
        [DataMember]
        public int Amount;
        [DataMember]
        public string CustomerName;
    }

    [DataContract]
    public class ShippingQuote
    {
        [DataMember]
        public int ShippingCost;
        [DataMember]
        public DateTime EstimatedShippingDate;
    }
}
